plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/impl/sql/AbstractSQLDialect.java

```diff
@@ -202,21 +204,25 @@ public TreeSet<String> getDataTypes(@Nullable DBPDataSource dataSource) {
         return types;
     }
 
+    @Nullable
     @Override
     public DBPKeywordType getKeywordType(@NotNull String word) {
-        return allKeywords.get(word.toUpperCase(Locale.ENGLISH));
+        for (Map.Entry<String, DBPKeywordType> entry : allKeywords.entrySet()) {
+            if (word.equalsIgnoreCase(entry.getKey())) {
+                return entry.getValue();
+            }
+        }
+        return null;
```

Comment:
This can slow things down, including formatting, autocompletion, and syntax highlighting.